frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/upgrade/CuratorStateStoreFilter.java

```diff
@@ -0,0 +1,60 @@
+package com.mesosphere.sdk.kafka.upgrade;
+
+import com.mesosphere.sdk.curator.CuratorStateStore;
+import com.mesosphere.sdk.curator.CuratorUtils;
+import com.mesosphere.sdk.offer.evaluate.placement.StringMatcher;
+import com.mesosphere.sdk.state.StateStoreException;
+import org.apache.curator.RetryPolicy;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * CuratorStateStoreFilter enables to ignore task names that match the filter.
+ */
+public class CuratorStateStoreFilter  extends CuratorStateStore {
```

Comment:
This is the most important class.

Original code makes lots of changes in SDK to make it compatible with upgrade (i.e. do not try to recover a task if it does not have a ServiceSpec, or do not count tasks for placement rules if they are not active and have no ServiceSpecs). This class is to reduce the number of changes necessary in SDK to support an upgrade.

Please see PR description.